generators/server/templates/src/main/java/package/service/UserService.java.ejs

```diff
@@ -911,12 +911,14 @@ public class UserService {
                     updateUser(user.getFirstName(), user.getLastName(), user.getEmail(),
                         user.getLangKey(), user.getImageUrl());
                 }
-                // no last updated info, blindly update
+            // no last updated info, blindly update
             } else {
                 log.debug("Updating user '{}' in local database", user.getLogin());
                 updateUser(user.getFirstName(), user.getLastName(), user.getEmail(),
                     user.getLangKey(), user.getImageUrl());
             }
+            // Add existing authorities stored in database to user
+            user.getAuthorities().addAll(existingUser.get().getAuthorities());
```

Comment:
It doesn't look appropriate. IDP manages the Roles/Groups association to a user account. On removing Role/Group association on the IDP side, we should also remove that from our app, and this synchronization code is trying to achieve the same.

Reply:
You're right. That part is probably to specific for our use case. I'll remove that.